### DISCUSSION TITLE
Nvidia driver dependencies and getting nvtop

### DISCUSSION THREAD

**Discussion by shotcollin** [filecoin-project/lotus#8201] [Storage Provider] [answered]
Has anyone else had trouble recently installing Nvidia drivers or utilities, particularly `nvtop`, without installing drivers from version 470? I'm running Ubuntu 20.04.3 'focal' with `linux-image-5.4.0-100-generic`, which I don't think you'd exactly call bleeding-edge, but I've noticed recently that installing any driver before 470 via `apt` repos, including backports, forces an upgrade to 470, as does installing `nvtop`, which then breaks drivers installed otherwise.

I've also tried compiling 460 from source, but that also didn't seem to offer a way to avoid going up to 470, even though `dpkg-shlibdeps` produced repeated messages like `warning: package could avoid a useless dependency if debian/libnvidia-extra-470/usr/lib/x86_64-linux-gnu/libnvidia-allocator.so.470.103.01 was not linked against libpthread.so.0 (it uses none of the library's symbols)`.

I'm fairly sure that dependency linking is something new, because I've previously installed the `nvidia-headless-460` and `nvidia-utils-460` packages from the Ubuntu repo (probably a dozen times) and had no issues with them or with getting `nvtop` to work with them. But now that same `apt` command installs driver version 470. And when using the `NVIDIA-Linux-x86_64-460.91.03.run` file to install the drivers, a newly installed `nvtop` has dependencies including `libnvidia-compute-470` and when installed it gives an error that it's `Impossible to initialize nvidia nvml : Driver/library version mismatch`. 

It's not like `nvtop` is some crucial piece of instrumentation or anything I can't happily survive without--I'm actually more wondering whether other people have noticed something like this or if it's just something I'm overlooking. I'm also wondering if this might be an indication that the 460 drivers are getting to an EOL stage that might make them more cumbersome to maintain if trying to keep the rest of the system relative up to date. Any thoughts?

**Comment by LexLuthr** [comment, marked as answer]
@shotcollin Were you able to figure out what was causing the problem?

**Reply by shotcollin** [reply]
@LexLuthr unfortunately, no, I didn't figure it out.

**Reply by LexLuthr** [reply]
https://bugs.launchpad.net/ubuntu/+source/nvtop/+bug/1835630
It is a bug with Ubuntu, it seems ubuntu team added this dependency and it has been there for a while. Maybe you can follow up there.

**Reply by shotcollin** [reply]
Interesting, thanks for looking into that @LexLuthr! I'm actually now running workers as containers with Debian as host OS and Debian nvtop works fine, but good to know nvtop in Ubuntu is still a possibility. Appreciate your finding the solution!
